src/components/features/ProgressBar.vue

```diff
@@ -0,0 +1,80 @@
+<template>
+  <div class="progress progress-striped">
+    <div class="progress__bar"></div>
+  </div>
+</template>
+
+<script>
+export default {
+  name: "ProgressBar",
+  props: ["startPercent", "config"],
+  data() {
+    return {
+      progressBar: null,
+      progressBarContainer: null
+    };
+  },
+
+  created() {
+    this.initializeProgressBar();
+  },
+
+  computed: {
+  },
+
+  methods: {
+    initializeProgressBar(){
+      setTimeout(() => {
+        this.progressBarContainer = document.querySelector(".progress")
+        this.progressBar = document.querySelector(".progress__bar")
+        this.setConfigProperties(this.config['details'])
+        this.progressTo(this.startPercent)
+      }, 200)
+    },
+
+    setConfigProperties(config){
+      // for the container
+      Object.assign(this.progressBarContainer.style, {
+        background: `${config['empty_color']}`
+      })
+
+      // for the bar
+      Object.assign(this.progressBar.style, {
+        "background-color": `${config['progress_color']}`
+      })
```

Comment:
assigning CSS according to the config passed. Right now it's just color, but if needed, more can be added

Reply:
Hmm... this has to be done in code, no? :( I'm sure there's a way to do it with separation, but I don't know :D